### PR TITLE
STD-35 | TheoryBlock commnads/queries

### PR DESCRIPTION
**Pull request by Evenya**
Ebis ono vse konem nada Meeting srochna

### CODE REVIEW

File: server/StudySharp.ApplicationServices/Commands/AddTheoryBlockCommand.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace StudySharp.ApplicationServices.Commands
+{
+    public sealed class AddTheoryBlockCommand : IRequest<OperationResult>
+    {
+        public string Name { get; set; }
+        public int CourseId { get; set; }
+    }
+
+    public sealed class AddTheoryBlockHandler : IRequestHandler<AddTheoryBlockCommand, OperationResult>
```
Comment (zefirlover) [inline server/StudySharp.ApplicationServices/Commands/AddTheoryBlockCommand.cs]: Change AddTheoryBlockHandler to AddTheoryBlockCommandHandler due to our name conventions

File: server/StudySharp.ApplicationServices/Commands/AddTheoryBlockCommand.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        private readonly StudySharpDbContext _context;
+
+        public AddTheoryBlockHandler(StudySharpDbContext sharpDbContext)
```
Comment (zefirlover) [inline server/StudySharp.ApplicationServices/Commands/AddTheoryBlockCommand.cs]: AddTheoryBlockCommandHandler

File: server/StudySharp.ApplicationServices/Commands/AddTheoryBlockCommand.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        {
+            _context = sharpDbContext;
+        }
+
+        public async Task<OperationResult> Handle(AddTheoryBlockCommand request, CancellationToken cancellationToken)
+        {
+            if (await _context.TheoryBlocks.AnyAsync(c => Equals(c.Name.ToLower()) && c.CourseId == request.CourseId, cancellationToken))
```
Comment (zefirlover) [inline server/StudySharp.ApplicationServices/Commands/AddTheoryBlockCommand.cs]: Change "c" to "_" due to our name conventions. AddCourseCommand.cs variant isn't correct

Comment (zefirlover) [inline server/StudySharp.ApplicationServices/Commands/AddTheoryBlockCommand.cs]: Where is request.Name.ToLower()?

File: server/StudySharp.ApplicationServices/Commands/AddTheoryBlockCommand.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            {
+                return OperationResult.Fail(string.Format(ErrorConstants.EntityAlreadyExists, nameof(TheoryBlock), nameof(TheoryBlock.Name), request.Name));
+            }
+
+            await _context.TheoryBlocks.AddAsync(
+                new TheoryBlock
+            {
+                Name = request.Name, CourseId = request.CourseId,
+            }, cancellationToken);
+            await _context.SaveChangesAsync(cancellationToken);
```
Comment (zefirlover) [inline server/StudySharp.ApplicationServices/Commands/AddTheoryBlockCommand.cs]: I think we need here an extra enter here, what do you think @KolaCation?

File: server/StudySharp.ApplicationServices/Commands/AddTheoryBlockCommand.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.VisualBasic;
+using StudySharp.Domain.Constants;
```
Comment (zefirlover) [inline server/StudySharp.ApplicationServices/Commands/AddTheoryBlockCommand.cs]: Are we really need here "System.Linq" and "Microsoft.VisualBasic" usings?

File: server/StudySharp.ApplicationServices/Commands/RemoveTheoryBlockByIdCommand.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        private readonly StudySharpDbContext _context;
+
+        public RemoveTheoryBlockByIdCommandHandler(StudySharpDbContext sharpDbContext)
+        {
+            _context = sharpDbContext;
+        }
+
+        public async Task<OperationResult> Handle(
+            RemoveTheoryBlockByIdCommand request,
+            CancellationToken cancellationToken)
```
Comment (zefirlover) [inline server/StudySharp.ApplicationServices/Commands/RemoveTheoryBlockByIdCommand.cs:27]: It's better to write that in one row

File: server/StudySharp.ApplicationServices/Queries/GetTheoryBlockByIdQuery.cs [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+using StudySharp.DomainServices;
+
+namespace StudySharp.ApplicationServices.Queries
+{
+    public sealed class GetTheoryBlockByIdQuery : IRequest<OperationResult<TheoryBlock>>
+    {
+        public int Id { get; set; }
+    }
+
+    // huita
```
Comment (zefirlover) [inline server/StudySharp.ApplicationServices/Queries/GetTheoryBlockByIdQuery.cs]: Remove the comment. Comments are banned to be in dev

**Review by zefirlover — DISMISSED** [review]
Change the name of the pull request. Pattern: STD-N | Name. (Copy it from Trello)

File: server/StudySharp.ApplicationServices/Queries/GetTheoryBlockByIdQuery.cs [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        private readonly StudySharpDbContext _context;
+
+        public GetTheoryBlockByIdQueryHandler(StudySharpDbContext studySharpDbContext)
+        {
+            _context = studySharpDbContext;
+        }
+
+        public async Task<OperationResult<TheoryBlock>> Handle(
+            GetTheoryBlockByIdQuery request,
+            CancellationToken cancellationToken)
```
Comment (zefirlover) [inline server/StudySharp.ApplicationServices/Queries/GetTheoryBlockByIdQuery.cs:27]: It's better to write that in one row

File: server/StudySharp.ApplicationServices/Commands/AddTheoryBlockCommand.cs [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+        private readonly StudySharpDbContext _context;
+
+        public AddTheoryBlockCommandHandler(StudySharpDbContext sharpDbContext)
+        {
+            _context = sharpDbContext;
+        }
+
+        public async Task<OperationResult> Handle(AddTheoryBlockCommand request, CancellationToken cancellationToken)
+        {
+            if (await _context.TheoryBlocks.AnyAsync(c => Equals(c.Name.ToLower(), request.Name.ToLower()) && c.CourseId == request.CourseId, cancellationToken))
```
Comment (zefirlover) [inline server/StudySharp.ApplicationServices/Commands/AddTheoryBlockCommand.cs]: Change "c" to "_"

File: server/StudySharp.ApplicationServices/Commands/AddCourseCommand.cs [hunk 10]
```diff
@@ -26,16 +26,16 @@ public AddCourseCommandHandler(StudySharpDbContext sharpDbContext)
...
                 return OperationResult.Fail(string.Format(ErrorConstants.EntityAlreadyExists, nameof(Course), nameof(Course.Name), request.Name));
             }
 
             await _context.Courses.AddAsync(
                 new Course
-                {
+            {
                     Name = request.Name, TeacherId = request.TeacherId,
-                }, cancellationToken);
+            }, cancellationToken);
```
Comment (zefirlover) [inline server/StudySharp.ApplicationServices/Commands/AddCourseCommand.cs]: Change it to the previous variant
And do the same at the AddTheoryBlockCommand

**Review by SleepyF0X — APPROVED** [review]

**Review by WerewolfNet — APPROVED** [review]

**Pull request merged**